conans/client/api/conan_api.py

```diff
@@ -218,27 +241,41 @@ def search_remote_recipes(self, remote, query):
         return data
 
     @api_method
-    def get_active_remotes(self, remote_names):
-        remotes = self.app.cache.registry.load_remotes()
-        all_remotes = remotes.all_values()
-
-        if not all_remotes:
-            raise ConanException("The remotes registry is empty. "
-                                 "Please add at least one valid remote.")
-
-        # If no remote is specified, search in all of them
-        if not remote_names:
-            # Exclude disabled remotes
-            all_remotes = [remote for remote in all_remotes if not remote.disabled]
-            return all_remotes
-
-        active_remotes = []
-        for remote_name in remote_names:
-            remote = remotes[remote_name]
-            active_remotes.append(remote)
+    def get_revisions(self, reference, is_package=False, remote=None):
```

Comment:
I've merged both logic for packages and recipes. I think it's not getting convoluted but let me know your opinion @tapia In my opinion, it would be a pain to have all those lines duplicated.

Reply:
I'm ok with the merging, but I don't like boolean flags (`is_package`), because it's not always obvious what they mean (you have to read the implementation to guess what is it if not a package).

I'd make two explicit API methods, calling an auxiliary function:

```python
def _get_revisions(self, reference, type_of_reference, remote):
    if type_of_reference == 'package':
        blahblah()
    elif type_of_reference == 'recipe':
        more_blah()

@api_method
def get_package_revisions(self, reference, remote):
    return self._get_revisions(reference, 'package', remote)
    
@api_method
def get_recipe_revisions(self, reference, remote):
    return self._get_revisions(reference, 'recipe', remote)
```

Reply:
Yes, agree, specially for API methods.
It could also implement the different type checking of ``reference`` outside of the ``_get_revisions()`` method.